src/stk/molecular/topology_graphs/topology_graph/optimizers/spinner.py

```diff
@@ -0,0 +1,146 @@
+"""
+Spinner
+=======
+
+"""
+
+from .optimizer import Optimizer
+from ..construction_state import ConstructionState
+
+import spindry as spd
+
+
+class Spinner(Optimizer):
+    """
+    Performs Monte Carlo optimisation of host-guest complexes [1]_.
+
+    Examples:
+        *Structure Optimization*
+
+        Using :class:`.Spinner` will lead to
+        :class:`.ConstructedMolecule` structures with better host-guest
+        structures. Especially useful for multiple-guest systems and
+        removing overlap.
+
+        .. testcode:: structure-optimization
+
+            import stk
+
+            bb1 = stk.BuildingBlock(
+                smiles='NCCN',
+                functional_groups=[stk.PrimaryAminoFactory()],
+            )
+            bb2 = stk.BuildingBlock(
+                smiles='O=CC(C=O)C=O',
+                functional_groups=[stk.AldehydeFactory()],
+            )
+            guest1 = stk.host_guest.Guest(
+                building_block=stk.BuildingBlock('c1ccccc1'),
+            )
+            guest2 = stk.host_guest.Guest(
+                building_block=stk.BuildingBlock('C1CCCCC1'),
+            )
+            cage = stk.ConstructedMolecule(
+                topology_graph=stk.cage.FourPlusSix(
+                    building_blocks=(bb1, bb2),
+                    optimizer=stk.MCHammer(),
+                ),
+            )
+
+            complex = stk.ConstructedMolecule(
+                topology_graph=stk.host_guest.Complex(
+                    host=stk.BuildingBlock.init_from_molecule(cage),
+                    guests=(guest1, guest2),
+                    optimizer=stk.Spinner(),
+                ),
+            )
+
+        Optimisation with :mod:`stk` simply collects the final position
+        matrix. The optimisation's trajectory can be output using the
+        :mod:`SpinDry` implementation if required by the user [1]_.
+        This code is entirely nonphysical and is, therefore, completely
+        general to any chemistry.
+
+    References
+    ----------
+    .. [1] https://github.com/andrewtarzia/SpinDry
+
+    """
+
+    def __init__(
+        self,
+        step_size: float = 1.5,
+        rotation_step_size: float = 5.,
+        num_conformers: int = 50,
+        max_attempts: int = 1000,
+        nonbond_epsilon: float = 5.,
+        nonbond_sigma: float = 1.2,
+        beta: float = 2.,
+        random_seed: int = 1000,
+    ) -> None:
+        """
+        Initialize an instance of :class:`.Spinner`.
+
+        Parameters:
+            step_size: The relative size of the step to take during
+                step.
+
+            rotation_step_size: The relative size of the rotation to
+                take during step.
+
+            num_conformers: Number of conformers to extract.
+
+            max_attempts: Maximum number of MC moves to try to generate
+                conformers.
+
+            nonbond_epsilon: Value of epsilon used in the nonbond
```

Comment:
Is it the non-bond potential or non-bonded potential? (or both?)

Reply:
I think nonbonded is better.

Reply:
How do you feel about chaning the argument name as well in that case?

Reply:
Poorly because that is the argument name in SpinDry and I would want consistency between the two.

Reply:
I think the argument/variable name is clear as to what it applies to.

Reply:
I don't think the consistency between the two is important. When stk wraps another libraries interface, it should not aim for consistency with it, because that would make stks own interface inconsistent. For example, stk cannot be both consistent with numpy, rdkit and SpindDry interfraces. it can only be consistent with its own.

I think it clear also, but if one form is better than the other, then I would prefer the better one, even if the first one is good.

Reply:
also stk has not control over the interface of its dependencies. This means spin dry might change its interface, and suddenly stk has a worse argument + its not even consistent any more.

Reply:
I agree with your statement, however, consistency with spindry is trivial compared to rdkit.

Additionally, `nonbond_epsilon` is used in the MCHammer implementation. So internal consistency is maintained.

Reply:
Yep, if stk already uses `nonbond_x` then I agree this should follow the same pattern.